x-pack/plugins/ml/public/application/timeseriesexplorer/components/entity_control/entity_control.tsx

```diff
@@ -136,6 +199,132 @@ export class EntityControl extends Component<EntityControlProps, EntityControlSt
+            ownFocus
+            button={
+              <EuiButtonIcon
+                color="text"
+                iconSize="xxl"
+                iconType="gear"
+                aria-label={i18n.translate('xpack.ml.timeSeriesExplorer.editControlConfiguration', {
+                  defaultMessage: 'Edit field configuration',
+                })}
+                onClick={() => {
+                  this.setState({
+                    isEntityConfigPopoverOpen: !this.state.isEntityConfigPopoverOpen,
+                  });
+                }}
+                data-test-subj={`mlSingleMetricViewerEntitySelectionConfigButton_${entity.fieldName}`}
+              />
+            }
+            isOpen={this.state.isEntityConfigPopoverOpen}
+            closePopover={() => {
+              this.setState({
+                isEntityConfigPopoverOpen: false,
+              });
+            }}
+          >
+            <EuiText>
+              <EuiFlexGroup gutterSize={'xs'} alignItems={'center'}>
+                <EuiFlexItem grow={false}>
+                  <EuiSwitch
+                    label={
+                      <FormattedMessage
+                        id="xpack.ml.timeSeriesExplorer.anomalousOnlyLabel"
+                        defaultMessage="Anomalous only"
+                      />
+                    }
+                    checked={!!this.props.config?.anomalousOnly}
+                    onChange={(e) => {
+                      const isAnomalousOnly = e.target.checked;
+                      this.props.onConfigChange(this.props.entity.fieldType, {
+                        anomalousOnly: isAnomalousOnly,
+                        sort: {
+                          order: this.props.config.sort.order,
+                          by: this.props.config.sort.by,
+                        },
+                      });
+                    }}
+                    compressed
+                    data-test-subj={`mlSingleMetricViewerEntitySelectionConfigAnomalousOnly_${entity.fieldName}`}
+                  />
+                </EuiFlexItem>
+
+                <EuiFlexItem grow={false} style={{ width: '16px', height: '24px' }}>
+                  {!this.props.config?.anomalousOnly ? (
+                    <EuiToolTip
+                      position="top"
+                      content={
+                        this.props.isModelPlotEnabled ? (
+                          <FormattedMessage
+                            id="xpack.ml.timeSeriesExplorer.nonAnomalousResultsWithModelPlotInfo"
+                            defaultMessage="You will be suggested to select values for all possible model plot results"
+                          />
+                        ) : (
+                          <FormattedMessage
+                            id="xpack.ml.timeSeriesExplorer.nonAnomalousResultsAllRecordsInfo"
+                            defaultMessage="You will be suggested to select values for all records created during the job lifetime"
```

Comment:
It seems to me that you just need to inform the user how this choice affects the contents of the drop-down list. So I'd suggest something like this:
```suggestion
                            defaultMessage="The list contains values from all records created during the lifetime of the job."
```

Reply:
Changed in https://github.com/elastic/kibana/pull/81510/commits/79d10e7bfcf867fd204ded3a31d1265bf3bd0788